components/ai_chat/resources/page/components/filter_menu/tools_menu.test.tsx

```diff
@@ -60,41 +89,164 @@ describe('tools_menu', () => {
+      })
+
+      const translateItem = screen
+        .getByText('translate')
+        .closest('leo-menu-item')
+      fireEvent.click(translateItem!)
+
+      expect(mockHandleSmartModeClick).toHaveBeenCalledWith(mockSmartModes[0])
+    })
+
+    it('should call handleSmartModeEdit when edit button is clicked', () => {
+      const mockHandleSmartModeEdit = jest.fn()
+      const { container } = renderWithContext(mockSmartModes, {
+        ...defaultProps,
+        handleSmartModeEdit: mockHandleSmartModeEdit,
+      })
+
+      const editButtons = container.querySelectorAll('leo-button')
+      const firstEditButton = editButtons[0] as HTMLElement
+      fireEvent.click(firstEditButton)
+
+      expect(mockHandleSmartModeEdit).toHaveBeenCalledWith(mockSmartModes[0])
+    })
+
+    it('should filter smart modes by shortcut', () => {
+      const { queryByText } = renderWithContext(mockSmartModes, {
+        ...defaultProps,
+        query: 'trans',
+      })
+
+      expect(queryByText('translate')).toBeInTheDocument()
+      expect(queryByText('explain')).not.toBeInTheDocument()
+    })
+
+    it('should show smart modes category when others hidden', () => {
+      const { getByText, queryByText } = renderWithContext(mockSmartModes, {
+        ...defaultProps,
+        query: 'translate',
+      })
+
+      expect(getByText('Smart Modes')).toBeInTheDocument()
+      expect(getByText('translate')).toBeInTheDocument()
+      expect(queryByText('category 1')).not.toBeInTheDocument()
+      expect(queryByText('category 2')).not.toBeInTheDocument()
+    })
+
+    it('should not render smart modes section when no handlers', () => {
+      const propsWithoutHandlers = {
+        ...defaultProps,
+        handleSmartModeClick: undefined,
+        handleSmartModeEdit: undefined,
+      }
+      const { queryByText } = renderWithContext(
+        mockSmartModes,
+        propsWithoutHandlers,
+      )
+
+      expect(queryByText('Smart Modes')).not.toBeInTheDocument()
+      expect(queryByText('translate')).not.toBeInTheDocument()
+    })
+
+    it('should render combined categories in correct order', () => {
+      const { container } = renderWithContext()
+
+      const menuItems = Array.from(container.querySelectorAll('leo-menu-item'))
+      const categoryHeaders = Array.from(
+        container.querySelectorAll('.menuSectionTitle, .menuSubtitle'),
+      )
+
+      // Smart Modes should appear first, then other categories
+      expect(categoryHeaders[0]).toHaveTextContent('Smart Modes')
+      expect(menuItems[0]).toHaveTextContent('translate')
+      expect(menuItems[1]).toHaveTextContent('explain')
+    })
+
+    it('should handle empty smart modes array', () => {
+      const { queryByText } = renderWithContext([])
+
+      expect(queryByText('Smart Modes')).not.toBeInTheDocument()
+      expect(queryByText('translate')).not.toBeInTheDocument()
+      expect(queryByText('explain')).not.toBeInTheDocument()
+    })
```

Comment:
I think we have a very similar test https://github.com/brave/brave-core/pull/31416#discussion_r2389477012?